tracer/src/Datadog.Trace/Configuration/IntegrationId.cs

```diff
@@ -57,5 +57,6 @@ internal enum IntegrationId
         HashAlgorithm,
         SymmetricAlgorithm,
         OpenTelemetry,
+        SpanProbe
```

Comment:
Consider adding an XML comment to clarify what this is and how it is different than the `TraceAnnotations` integration?

Reply:
What are the implications if someone disables `SpanProbe`? _Should_ they be able to 🤔